File: NumberFormattingSample/js/js/scenario2_decimalformatting.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";
    var page = WinJS.UI.Pages.define("/html/scenario2_DecimalFormatting.html", {
        ready: function (element, options) {
            document.getElementById("displayButton").addEventListener("click", doDisplay, false);
        }
    });

    function doDisplay() {
        // This scenario uses the Windows.Globalization.NumberFormatting.DecimalFormatter class
        // to format a number.

        // Create formatter initialized using the current user's preference settings.
        var decimalFormat = new Windows.Globalization.NumberFormatting.DecimalFormatter();

        // Make a random number.
        var randomNumber = (Math.random() * 100000);

        // Format with the user's default preferences.
        var decimal = decimalFormat.format(randomNumber);

        // Format with grouping.
        var decimalFormat1 = new Windows.Globalization.NumberFormatting.DecimalFormatter();
        decimalFormat1.isGrouped = true;
        var decimal1 = decimalFormat1.format(randomNumber);

        // Format with grouping using French.
        var decimalFormatFR = new Windows.Globalization.NumberFormatting.DecimalFormatter(["fr-FR"], "ZZ");
        decimalFormatFR.isGrouped = true;
        var decimalFR = decimalFormatFR.format(randomNumber);

        // Illustrate how automatic digit substitution works
        var decimalFormatAR = new Windows.Globalization.NumberFormatting.DecimalFormatter(["ar"], "ZZ");
        decimalFormatAR.isGrouped = true;
        var decimalAR = decimalFormatAR.format(randomNumber);

        // Get a fixed number.
        var fixedNumber = 500;

        // Format with the user's default preferences.
        var decimal2 = decimalFormat.format(fixedNumber);

        // Format always with a decimal point.
        var decimalFormat3 = new Windows.Globalization.NumberFormatting.DecimalFormatter();
        decimalFormat3.isDecimalPointAlwaysDisplayed = true;
        decimalFormat3.fractionDigits = 0;
        var decimal3 = decimalFormat3.format(fixedNumber);

        // Format with no fractional digits or decimal point.
        var decimalFormat4 = new Windows.Globalization.NumberFormatting.DecimalFormatter();
        decimalFormat4.fractionDigits = 0;
        var decimal4 = decimalFormat4.format(fixedNumber);

        // Display the results.
        var results = "Random number (" + randomNumber + ")\n" +
                      "With current user preferences: " + decimal + "\n" +
                      "With grouping separators: " + decimal1 + "\n" +
                      "With grouping separators (fr-FR): " + decimalFR + "\n" +
                      "With digit substitution (ar): " + decimalAR + "\n\n" +
                      "Fixed number (" + fixedNumber + ")\n" +
                      "With current user preferences: " + decimal2 + "\n" +
                      "Always with a decimal point: " + decimal3 + "\n" +
                      "With no fraction digits or decimal points: " + decimal4;

        document.getElementById("output").innerText = results;
    }
})();

File: Samples/NumberFormatting/cppwinrt/Scenario2_DecimalFormatting.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario2_DecimalFormatting.h"
#include "Scenario2_DecimalFormatting.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::Globalization::NumberFormatting;
using namespace Windows::UI::Xaml;

namespace winrt::SDKTemplate::implementation
{
    Scenario2_DecimalFormatting::Scenario2_DecimalFormatting()
    {
        InitializeComponent();
    }

    void Scenario2_DecimalFormatting::Display_Click(IInspectable const&, RoutedEventArgs const&)
    {
        // This scenario uses the Windows.Globalization.NumberFormatting.DecimalFormatter class
        // to format a number.

        // Create formatter initialized using the current user's preference settings.
        DecimalFormatter decimalFormat;

        // Make a random number.
        float randomNumber = (float)rand() / RAND_MAX * 100000.0f;

        // Format with the user's default preferences.
        hstring decimal = decimalFormat.Format(randomNumber);

        // Format with grouping.
        DecimalFormatter decimalFormat1;
        decimalFormat1.IsGrouped(true);
        hstring decimal1 = decimalFormat1.Format(randomNumber);

        // Format with grouping using French.
        DecimalFormatter decimalFormatFR({ L"fr-FR" }, L"ZZ");
        decimalFormatFR.IsGrouped(true);
        hstring decimalFR = decimalFormatFR.Format(randomNumber);

        // Illustrate how automatic digit substitution works
        DecimalFormatter decimalFormatAR({ L"ar" }, L"ZZ");
        hstring decimalAR = decimalFormatAR.Format(randomNumber);

        // Get a fixed number.
        long long fixedNumber = 500;

        // Format with the user's default preferences.
        hstring decimal2 = decimalFormat.Format(fixedNumber);

        // Format always with a decimal point.
        DecimalFormatter decimalFormat3;
        decimalFormat3.IsDecimalPointAlwaysDisplayed(true);
        decimalFormat3.FractionDigits(0);
        hstring decimal3 = decimalFormat3.Format(fixedNumber);

        // Format with no fractional digits or decimal point.
        DecimalFormatter decimalFormat4;
        decimalFormat4.FractionDigits(0);
        hstring decimal4 = decimalFormat4.Format(fixedNumber);

        // Display the results.
        std::wostringstream results;
        results <<
            L"Random number (" << randomNumber << L")" << std::endl <<
            L"With current user preferences: " << std::wstring_view(decimal) << std::endl <<
            L"With grouping separators: " << std::wstring_view(decimal1) << std::endl <<
            L"With grouping separators (fr-FR): " << std::wstring_view(decimalFR) << std::endl <<
            L"With digit substitution (ar): " << std::wstring_view(decimalAR) << std::endl <<
            std::endl <<
            L"Fixed number (" << fixedNumber << L")" << std::endl <<
            L"With current user preferences: " << std::wstring_view(decimal2) << std::endl <<
            L"Always with a decimal point: " << std::wstring_view(decimal3) << std::endl <<
            L"With no fraction digits or decimal points: " << std::wstring_view(decimal4);

        OutputTextBlock().Text(results.str());
    }
}
